### PR TITLE
Create handling-exception-notes.md

### PR DESCRIPTION
**Pull request by MaxenceGui**
## Pull request to add handling and raising exceptions in Python training

*  Reference issue #113

- [x] Read and take note on [Real Python Exception Tutorial](https://realpython.com/python-raise-exception/#chaining-exceptions-with-the-from-clause)
- [x] Write the documentation that will guide CFIA Handling and Raising Exception
- [x] Modify TRAINING.md file to add the documentation to the backend developer training

### CODE REVIEW

File: handling-exception-training.md [hunk 1]
```diff
@@ -0,0 +1,1064 @@
...
+...     raise
+
+ERROR:root:division by zero
+Traceback (most recent call last):
+  File "<stdin>", line 5, in <module> # will only appear if 'raise error' was called
+  File "<stdin>", line 2, in <module>
+ZeroDivisionError: division by zero
+```
+
+#### Les Exceptions de groupes
```
Comment (rngadam) [inline handling-exception-training.md:705]: when should we be using group exception? do we expect our developers to use this?

I'd like to have a documentation focused on our practices, not every possible variation.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: handling-exception-notes.md [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+## Before starting
+
+🚧 This files is not intended to be push to the main branch of dev-rel-docs.
+It's just to give a chance for everyone to review the note I took 🚧
+
+- [Version Française](#avant-de-commencer) :small_blue_diamond:
+- This training is aimed at the backend developer, but everybody is welcome to
+  read it.
+
+|:wave: To know more| |:--| |[Pozo Ramos, L. (2023,19 juin). *Python's raise:
```
Comment (ibrahim-kabir) [inline handling-exception-notes.md:12]: This emoji and the table does not display on GitHub when I enable the "Display rich diff" option.
image

File: handling-exception-notes.md [hunk 3]
```diff
@@ -0,0 +1,652 @@
+# Python: Handling and Raising Exception / Gérer et soulever des erreurs
+
+## Before starting
+
+🚧 This files is not intended to be push to the main branch of dev-rel-docs.
```
Comment (ibrahim-kabir) [inline handling-exception-notes.md:5]: typo
```suggestion
🚧 This file is not intended to be push to the main branch of dev-rel-docs.
```

File: handling-exception-notes.md [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+
+```
+
+#### Link to documentation for different exception types
+
+|:warning:|:collision:|:interrobang:|
+|--|--|--|
+|[Warning](https://docs.python.org/3/library/warnings.html#warning-categories)|[Error](https://docs.python.org/3/library/exceptions.html#concrete-exceptions)|[Exception](https://docs.python.org/3/library/exceptions.html#Exception)|
+
+Every exception have a [traceback](https://realpython.com/python-traceback/)
```
Comment (ibrahim-kabir) [inline handling-exception-notes.md:48]: typo

```suggestion
Every exception has a [traceback](https://realpython.com/python-traceback/)
```

File: handling-exception-notes.md [hunk 5]
```diff
@@ -0,0 +1,652 @@
...
+
+```python
+colors = ["red", "blue", "green"]
+
+# Basic handling of error
+
+try:
+    colors[10]
+except: IndexError:
+    print("Your list doesn't have that index :-(")
```
Comment (ibrahim-kabir) [inline handling-exception-notes.md:65]: syntax error

```suggestion
try:
    colors[10]
except IndexError:
    print("Your list doesn't have that index :-(")
```

File: handling-exception-notes.md [hunk 6]
```diff
@@ -0,0 +1,652 @@
...
+TypeError: number expected, got str
+
+The above exception was the direct cause of the following exception:
+
+Traceback (most recent call last):
+  File "<stdin>", line 4, in <module>
+ValueError: invalid argument
+```
+
+> If your note using from the message will be different. Without `from`, the
```
Comment (ibrahim-kabir) [inline handling-exception-notes.md:459]: typo

```suggestion
> If you're not using `from`, the message will be different. Without `from`, the
```

**Review by ibrahim-kabir — CHANGES_REQUESTED** [review]

File: handling-exception-notes.md (hunk 2 above)
Comment (MaxenceGui) [reply handling-exception-notes.md:12]: Done

File: handling-exception-notes.md (hunk 3 above)
Comment (MaxenceGui) [reply handling-exception-notes.md:5]: Done

Comment (RussellJimmies) [reply handling-exception-notes.md:5]: to be pushed

File: handling-exception-notes.md (hunk 4 above)
Comment (MaxenceGui) [reply handling-exception-notes.md:48]: Done

File: handling-exception-notes.md (hunk 5 above)
Comment (MaxenceGui) [reply handling-exception-notes.md:65]: Done

**Review by RussellJimmies — CHANGES_REQUESTED** [review]

File: handling-exception-notes.md (hunk 6 above)
Comment (MaxenceGui) [reply handling-exception-notes.md:459]: Done

File: handling-exception-notes.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (MaxenceGui) [reply handling-exception-notes.md:5]: Done

File: handling-exception-notes.md [hunk 7]
```diff
@@ -0,0 +1,652 @@
+# Python: Handling and Raising Exception / Gérer et soulever des erreurs
+
+## Before starting
+
+🚧 This files is not intended to be push to the main branch of dev-rel-docs.
+It's just to give a chance for everyone to review the note I took 🚧
```
[the remark this answers is not in the document]
Comment (MaxenceGui) [reply handling-exception-notes.md:6]: Done

File: handling-exception-notes.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (MaxenceGui) [reply handling-exception-notes.md:48]: Done

File: handling-exception-notes.md [hunk 8]
```diff
@@ -0,0 +1,652 @@
...
+class MyException(Exception)
+    pass
+
+raise MyException("an error occurred")
+```
+
+> The `from` clause is optional. It allows the developers to chain exceptions
+> together.
+
+|:warning:| Raising `Exception` instance is not considered best practice.|
```
[the remark this answers is not in the document]
Comment (MaxenceGui) [reply handling-exception-notes.md:110]: Done

File: handling-exception-notes.md [hunk 9]
```diff
@@ -0,0 +1,652 @@
...
+```
+
+#### Link to documentation for different exception types
+
+|:warning:|:collision:|:interrobang:|
+|--|--|--|
+|[Warning](https://docs.python.org/3/library/warnings.html#warning-categories)|[Error](https://docs.python.org/3/library/exceptions.html#concrete-exceptions)|[Exception](https://docs.python.org/3/library/exceptions.html#Exception)|
+
+Every exception have a [traceback](https://realpython.com/python-traceback/)
+attribute, also name:
```
[the remark this answers is not in the document]
Comment (MaxenceGui) [reply handling-exception-notes.md:49]: Done

**Review by ibrahim-kabir — APPROVED** [review]

**Review by k-allagbe — APPROVED** [review]
Informative. Never really paid attention to exception message formatting before. 👍

**Review by rngadam — CHANGES_REQUESTED** [review]
Suggest removing the -notes.md and splitting the french and english into two markdowns (handling-exception-training.md and handling-exception-training.fr-ca.md) as it's a bit unwieldy to have both in the same file.

**Comment by MaxenceGui** [comment]
> Suggest removing the -notes.md and splitting the french and english into two markdowns (handling-exception-training.md and handling-exception-training.fr-ca.md) as it's a bit unwieldy to have both in the same file.

Implement your change

**Review by rngadam — APPROVED** [review]

**Comment by MaxenceGui** [comment]
@RussellJimmies if it's ok with you, we can merge these files and open an issue to review and correct their grammar since they are big files.

**Review by RussellJimmies — APPROVED** [review]

**Pull request merged**